Source/WTF/wtf/haiku/RunLoopHaiku.cpp

```diff
@@ -62,6 +76,42 @@ RunLoop::RunLoop()
 	: m_looper(nullptr)
 {
     m_handler = new LoopHandler();
+
+    // We want to be able to queue messages before the RunLoop::run has been
+    // called, so we need to attach our handler to a looper now. However, we
+    // don't need to start the looper yet.
+
+    BLooper* looper;
+
+    BLooper* currentLooper = BLooper::LooperForThread(find_thread(NULL));
+    if (currentLooper) {
+        // This thread already has a looper (likely the BApplication looper).
+        // Attach our handler to it.
+        looper = currentLooper;
+    } else {
+        // I'm not sure how this code would behave if you created a team without
+        // a BApplication. Comment this ASSERT out and give it a try if you
+        // desire.
+        ASSERT(be_app);
+
+        thread_info main_thread;
+        int32 cookie = 0;
+        get_next_thread_info(0, &cookie, &main_thread);
+        if (be_app->Thread() == B_ERROR && find_thread(NULL) == main_thread.thread) {
```

Comment:
I'm not sure you need to check be_app->Thread here. If we are on the main thread, let's try to use be_app.